.github/workflows/github-actions.yml

```diff
@@ -0,0 +1,63 @@
+name: Tests
+on:
+  push:
+    branches: add_test
+  pull_request:
+    branches: main
+
+jobs:
+  unit-testing:
+    defaults:
+      run:
+        # Set default shell to login-bash
+        # This is required to properly activate the conda environment
+        # https://github.com/conda-incubator/setup-miniconda?tab=readme-ov-file#use-a-default-shell
+        shell: bash -el {0}
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    
+    steps:
+      - name: Install xmllint on Linux
+        if: runner.os == 'Linux'
+        run: sudo apt-get install libxml2-utils
+            
+      - name: Get safe PYTHON_VERSION
+        run:  echo "PYTHON_VERSION=$(curl --location https://devguide.python.org/versions | xmllint --html --xpath '//section[@id="supported-versions"]//table/tbody/tr[count(//section[@id="supported-versions"]//table/tbody/tr[td[.="security"]]/preceding-sibling::*)]/td[1]/p/text()' - 2> /dev/null)" >> $GITHUB_ENV
+  
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Update environment.yml with PYTHON_VERSION
+        run: sed 's/python.*/python==${{ env.PYTHON_VERSION }}/' environment.yml > environment_new.yml
```

Comment:
I think that your regex is too loose. I would be more specific with:

```
sed -r 's/- python[>=]+[0-9]+\.[0-9]+/- python==${{ env.PYTHON_VERSION }}/
```

1. `-r` uses "regex extended" as `sed` by default uses "basic regular expression" (BRE), which is very limited
2. match on `- python` rather than just `python` for specificity
3. `[>=]+` matches "one or more characters found in the set '(>=)'"
4. `[0-9]+` matches "one or more digits"

Reply:
> your regex is too loose

Agree. I was just searching for ANY python. IIUC, your concern is for a scenario where this can go wrong. For instance, if another package is added that has `python` as part of its name.  Thanks for the note 👍  

> -r uses "regex extended" as sed by default uses "basic regular expression" (BRE),

Good to know 👍

Reply:
> IIUC, your concern is for a scenario where this can go wrong. For instance, if another package is added that has python as part of its name.

Yes, exactly!